go/vt/vttablet/tabletserver/txlogz.go

```diff
@@ -120,22 +120,26 @@ func txlogzHandler(w http.ResponseWriter, req *http.Request) {
 				log.Error(err)
 				continue
 			}
-			duration := txc.txProps.EndTime.Sub(txc.txProps.StartTime).Seconds()
-			var level string
-			if duration < 0.1 {
-				level = "low"
-			} else if duration < 1.0 {
-				level = "medium"
-			} else {
-				level = "high"
-			}
-			tmplData := struct {
-				*StatefulConnection
-				Duration   float64
-				ColorLevel string
-			}{txc, duration, level}
-			if err := txlogzTmpl.Execute(w, tmplData); err != nil {
-				log.Errorf("txlogz: couldn't execute template: %v", err)
+			props := txc.txProps
+			// not all StatefulConnections contain transactions
+			if props != nil {
```

Comment:
wanna do early return?

```suggestion
			if props == nil {
			continue
```